src/k8s/pkg/utils/cidr.go

```diff
@@ -142,3 +142,21 @@ func ToIPString(ip net.IP) string {
 	}
 	return "[" + ip.String() + "]"
 }
+
+// CIDRsOverlap checks if two given CIDR blocks overlap.
+// It takes two strings representing the CIDR blocks as input and returns a boolean indicating
+// whether they overlap and an error if any of the CIDR blocks are invalid.
+func CIDRsOverlap(cidr1, cidr2 string) (bool, error) {
```

Comment:
I'd just inline this logic (90% of this function is just `net.ParseCIDR`) but I'm also fine with keeping this.

Reply:
Since this function hides 2 calls to net.Parse, and we call this function 2 times, it means I would need to add 4 calls to net.Parse and their associated error handling in validateCIDROverlap, which will bloat the function.